src/adapters/IDB.js

```diff
@@ -74,6 +74,7 @@ export async function execute(db, name, callback, options = {}) {
     }
     transaction.onerror = event => reject(event.target.error);
     transaction.oncomplete = event => resolve(result);
+    transaction.onabort = () => reject(transaction.error);
```

Comment:
This property is null if the transactionwas aborted with the IDBTransaction.abort method.

[source](https://developer.mozilla.org/en-US/docs/Web/API/IDBTransaction/error)

Reply:
> All pending IDBRequest objects created during this transaction have their IDBRequest.error attribute set to AbortError.

[source](https://developer.mozilla.org/en-US/docs/Web/API/IDBTransaction/abort)

Reply:
Oh, interesting! That's how they did it in https://phabricator.services.mozilla.com/D36822  I didn't even check the docs...

Reply:
However I have no idea how to grab the IDBRequest at this point.
Maybe the error is linked to the event:

```suggestion
    transaction.onabort = () => reject(event.target.error);
```

Reply:
Well, like the error then `transaction.onabort = event => reject(event.target.error);`

Reply:
We shall try both and see what happens.

Reply:
@leplatrem @Natim I just took another look to what is the error rejected (if any) on the test case that was getting stuck without this fix applied in IndexedDB.jsm and in that case `transaction.error` was set to `QuotaExceededError`.

Then I tried to see what is happening when I explicitly reject the transaction using transaction abort and for me both `transaction.error` and `event.target.error` were not defined.

I haven't yet submitted a patch for IndexedDB.jsm on phabricator, but I'm thinking to resort to something like the following to be sure to always reject the transaction with an error:

```
transaction.onabort = (evt) => {
  let error =
    transaction.error ||
    evt.target.error ||
    new DOMException("The operation has been aborted", "AbortError");
  reject(error);
};
```

I'll keep an eye on this PR (and let you know once I have further updates about how we are going to handle this in the wrapper defined by IndexedDB.jsm).

Reply:
(I just filed [Bug 1573828](https://bugzilla.mozilla.org/1573828) for tracking the issue for  IndexedDB.jsm)